### PR TITLE
Homework week 4 ChrisFayad

### PR DESCRIPTION
**Pull request by ChrisFayad**

### CODE REVIEW

File: 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js [hunk 1]
```diff
@@ -12,12 +12,9 @@ Let's rewrite it (or _refactor_ it, as experienced developers would call it):
 // ! Function to be tested
 function doubleEvenNumbers(numbers) {
   // TODO rewrite the function body using `map` and `filter`.
-  const newNumbers = [];
-  for (let i = 0; i < numbers.length; i++) {
-    if (numbers[i] % 2 === 0) {
-      newNumbers.push(numbers[i] * 2);
-    }
-  }
+  const newNumbers = numbers
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js:15]: Well done!

File: 1-JavaScript/Week4/homework/ex2-mondaysWorth.test.js [hunk 2]
```diff
@@ -31,8 +31,13 @@ const mondayTasks = [
 
 const hourlyRate = 25;
 
-function computeEarnings(/* TODO parameter(s) go here */) {
+function computeEarnings(tasks, rating) {
   // TODO complete this function
+  const earning = tasks
+    .map( ({duration}) => duration / 60)
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex2-mondaysWorth.test.js:37]: I'd not map through the array twice. You can already in the first loop get the value that you want. You can only do in 2 steps inside the map, if you want that for clarity and readability:
```
.map( ({duration}) => {
   const taskEarn = duration / 60;
  return taskEarn * rating;
})
```

File: 1-JavaScript/Week4/homework/ex2-mondaysWorth.test.js [hunk 3]
```diff
@@ -31,8 +31,13 @@ const mondayTasks = [
...
+    .map(taskEarn => taskEarn * rating)
+    .reduce((previous, current) => previous + current ,0)
+  return `€${earning.toFixed(2)}`
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex2-mondaysWorth.test.js:40]: Well done!

File: 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js [hunk 4]
```diff
@@ -26,26 +26,25 @@ const fruitBasket = [
 ];
 
 // ! Function under test
-function sanitizeFruitBasket(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function sanitizeFruitBasket(fruitBasket, lemon) {
+  const newBasket = fruitBasket.filter(fruit => fruit !== lemon);
+  return newBasket
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js:31]: Watch out for the `;` at the end of the lines when needed.

File: 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js [hunk 5]
```diff
@@ -26,26 +26,25 @@ const fruitBasket = [
...
-    // TODO replace next line with your code
-    expect(false).toBe(true);
+    sanitizeFruitBasket(fruitBasket, 'lemon');
+    expect(originalFruitBasketContents).toEqual(fruitBasket);
   });
 
   test('should return a new array that does not include the unwanted `lemon`', () => {
-    // TODO replace next line with your code
-    expect(false).toBe(true);
+    expect(sanitizeFruitBasket(fruitBasket, 'lemon')).not.toContain('lemon');
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js:48]: Well done the tests!!

File: 1-JavaScript/Week4/homework/ex4-observable/ex4-observable.js [hunk 6]
```diff
@@ -18,9 +18,11 @@ function createObservable() {
   return {
     subscribe: function (subscriber) {
       // TODO complete this function
+      subscribers.push(subscriber);
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex4-observable/ex4-observable.js:21]: Yes, well solved!

**Comment by killoncio** [comment]
Well done, Chris!

Please, only add to each merge requests the changes for the corresponding exercises. This one is called 'week 4', but it contains also the exercises for week 3. I did not check them, since I'm doing that in your previous merge request.

Keep it up with the good work!! :)

File: 1-JavaScript/Week4/homework/ex2-mondaysWorth.test.js (hunk 2 above)
Comment (ChrisFayad) [reply 1-JavaScript/Week4/homework/ex2-mondaysWorth.test.js:37]: Thanks for the feedback, I will implement that in future codes!

**Pull request closed without merging**